### PR TITLE
Add external CAVP tests for HKDF_expand as a KDF in Feedbacak Mode

### PR DESCRIPTION
**Pull request by Taffer**
### Issues:
Resolves CryptoAlg-1436

### Description of changes: 
Adds NIST CAVP known-answer tests for using `HKDF_expand()` as a KDF in Feedback Mode, per NIST SP 800-108r1. This is similar to the SSHKDF CAVP tests that were recently added.

### Call-outs:
Previously `HKDF_expand()` didn't have any specific tests, it was being tested as part of the `HKDF()`.

### Testing:
This adds 125 ms to the `crypto_test` unit tests on my system.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and  the ISC license.

### CODE REVIEW

**Review by dkostic — DISMISSED** [review]

File: crypto/fipsmodule/hkdf/hkdf_test.cc [hunk 1]
```diff
@@ -416,3 +416,166 @@ TEST(HKDFTest, WycheproofSHA512) {
...
+
+  if ((ctrlocation != "AFTER_FIXED") || (rlen != "8_BITS")) {
+    // HKDF_expand() only works for "Counter After" and an r length of 8 bits.
+    // Skip this test.
+    run_test = false;
+  }
+
+  const EVP_MD *md = NULL;
+  if (prf == "HMAC_SHA1") {
+    // Coverity will yell at us for this, as it should. But FIPS 140-3 still
```
[the remark this answers is not in the document]
Comment (Taffer) [reply crypto/fipsmodule/hkdf/hkdf_test.cc:448]: As part of the FIPS team, I prefer to take a more prescriptive approach, and would like people to stop using broken crypto... IMHO we should be more opinionated about this in AWS-LC.

[the remark this answers is not in the document]
Comment (dkostic) [reply crypto/fipsmodule/hkdf/hkdf_test.cc:448]: I agree that we should be more opinionated, but HMAC with SHA1 is perfectly fine, there is no security issue here, as someone might conclude based on the comment. Maybe just add: "But FIPS 140-3 still allows it in some circumstances where using SHA1 is secure, for example, in HMAC.". Anyway, not the hill I'm willing to die on :)

**Review by dkostic — APPROVED** [review]

**Comment by Taffer** [comment]
Looks like I'm going to have to throw out this PR and make a new one because of the supposed `crypto_test_data.cc` conflict. Current build is rebased off the tip of `aws/aws-lc`'s `main`, still a conflict. 🤷

**Comment by Taffer** [comment]
Closing this in  favour of PR 840 due to the `generated-src` conflict.

**Pull request closed without merging**